File: 0003 longest-substring-without-repeating-characters/solution.go
func lengthOfLongestSubstring(s string) int {
    ma:=make(map[byte]int)
    l,m:=0,0
    for r:=0;r<len(s);r++ {
        if _,ok :=ma[s[r]]; ok {
            l=max(l,ma[s[r]]+1)
        }
        ma[s[r]]=r
        m=max(m,r-l+1)
    }
    return m
}

File: 0003 longest-substring-without-repeating-characters/solution.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        boolean[] map = new boolean[128];
        int l=0,gap=0;
        for(int r=0;r<s.length();r++) {
            char c = s.charAt(r);
            while(l<=r && map[c]) {
                map[s.charAt(l)]=false;
                l++;
            }
            map[c]=true;
            gap=Math.max(gap,r-l+1);
        }
        return gap;
    }
}
